elements/lisk-chain/test/integration/state_store/save_diff.spec.ts

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import * as path from 'path';
```

Comment:
does this need to be integration test?

Reply:
for db we need the file path know.

Reply:
i meant this whole test suite. does it need to be tested with compatibility of database?
I thought it is enough to mock the DB and test the behavior

Reply:
As discussed, no other test covers diff saving into the database, so it is valid to be in integration